### PR TITLE
implement multiproofs

### PR DESCRIPTION
**Pull request by ralexstokes**
implements basic verification logic for single and multi proofs using generalized indices

TODO:
- [x] make pass on error handling and following existing discipline around panics etc

FUTURE:
implement type-safe generalized index paths to generalized index
implement proving code to generate proofs in first place

### CODE REVIEW

File: ssz-rs/src/merkleization/multiproofs/generalized_index.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+// From: https://users.rust-lang.org/t/logarithm-of-integers/8506/5
+const fn num_bits<T>() -> usize {
+    std::mem::size_of::<T>() * 8
```
Comment (ec2) [inline ssz-rs/src/merkleization/multiproofs/generalized_index.rs:3]:
```suggestion
    core::mem::size_of::<T>() * 8
```
For `no_std` compatible.

File: ssz-rs/src/merkleization/multiproofs/mod.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
+mod generalized_index;
+
+pub use generalized_index::*;
+
+use crate::merkleization::Node;
+use sha2::{Digest, Sha256};
+use std::collections::{HashMap, HashSet};
```
Comment (ec2) [inline ssz-rs/src/merkleization/multiproofs/mod.rs:7]: Use BTreeSet and BTreeMap in `alloc` crate for no_std

Comment (ralexstokes) [reply ssz-rs/src/merkleization/multiproofs/mod.rs:7]: thanks! I may not merge this in right now but ill make an issue at the least

**Comment by ralexstokes** [comment]
thanks @ec2 ! went ahead and make it `no-std` compatible :)

**Pull request merged**